src/bootstrap/bin/main.rs

```diff
@@ -5,9 +5,13 @@
 //! parent directory, and otherwise documentation can be found throughout the `build`
 //! directory in each respective module.
 
+use std::env;
+#[cfg(all(any(unix, windows), not(target_os = "solaris")))]
 use std::fs::OpenOptions;
```

Comment:
`OpenOptions` only used once, so you could consider using it as `std::fs::OpenOptions` instead. So we can avoid conditional import for this which makes the code a bit more difficult to read.